scps/SCP-6.md

```diff
@@ -0,0 +1,134 @@
+
+Skipping the version field `v`, which is as barebones as it gets, and moving onto `id`. This field can be set and used by Ipfs-log users to identify and separate entries made to different logs, it is meant to be a string. There is not much to change here as it's just a string set by the user in userland (like in OrbitDB). However in entry version 3 `id` is renamed to `log` because id is used in a couple different places and log seems much better at describing the property.
+
+The `clock` field's value is an object with two properties, `time` and `id`. It is a Lamport clock logical timestamp and is the heaviest ordering metric of entries by default. The `clock.time` field, like `v`, is simply a positive integer, however `clock.id` by default is set to identity.publicKey which already exists in other places in the entry. This is a good chunk of bytes that can be excluded, either by lazily adding `clock.id` when explicitly set by the user or removing it completely (changing `clock` to what `clock.time` is now).
+
+Fields relating to writer authentication are where the bulk of the bytes can be trimmed from the entry. The `key` field contains the writer's publicKey from their identity. This publicKey also exists at `identity.publicKey` so `key` could be removed from the entry.
+
+As mentioned in [The Road to 1.0](https://github.com/orbitdb/orbit-db/issues/819), `separate identities/keys from oplog entry. CID per identity/key. cuts N bytes from each pubsub message and bitswap/ipfs/ild block transfer.`, the `identity` field could be replaced with the CID of the identity. This deduplication should work well as the `identity` field is the same data in every entry for each unique writer. However in situations where writer keys are ephemeral this is less advantaged and may be a good area to leave flexible.
+
+The `sig` field contains a string, hex encoded digital signature of fields: `v`, `clock`, `id`, `payload`, `next`, `refs`, and `hash`. It is used to verify that `identity` is the one that created the entry and that it wrote the entry. Since the data is currently a hex encoded string, it could instead be encoded into base64 strings to pinch a few bytes (~40B per `sig`). Base64 uses [33% less space](https://stackoverflow.com/questions/3183841/base64-vs-hex-for-sending-binary-content-over-the-internet-in-xml-doc) than base16/hex so if encoding/decoding speeds are similar this change may make sense.
+
+Entry version 3 does not propose any changes to fields `payload`, `next`, or `refs`.
+
+### Entry version 3
+
+What's changed?

```

Comment:
Great work @tabcat! 👏  I'm liking where this is going.

I'd like to discuss a couple of changes to this, see the suggestions below.